# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build ffmpeg-build gst-plugins-base-build"

PKG_CONFIG_PATH=/wl/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.16; lt-2.4.6)
libtoolize -c -f; aclocal -I m4 -I common/m4; autoconf; autoheader -f; automake -a -c
rm -rf gst-libs/ext
PKG_CONFIG_PATH=/wl/lib/pkgconfig LDFLAGS=-Wl,-rpath,/wl/lib ./configure --with-system-libav --prefix=/wl

# build
make

# install
install -d $DESTDIR/wl/lib/gstreamer-1.0
install ext/libav/.libs/libgstlibav.so $DESTDIR/wl/lib/gstreamer-1.0

# build.pc
install -d $DESTDIR/wl/share/pkgconfig
cat > $DESTDIR/wl/share/pkgconfig/gst-libav-build.pc << EOF
Name: gst-libav
Version: 1.14.5
Description: FFmpeg GStreamer plugin
Requires: $REQUIRES

exec=\\
/wl/lib/gstreamer-1.0/libgstlibav.so
EOF

exit
--- gst-libav-1.14.5.orig/configure.ac
+++ gst-libav-1.14.5/configure.ac
@@ -499,7 +499,6 @@
 ext/libav/Makefile
 ext/libswscale/Makefile
 gst-libs/Makefile
-gst-libs/ext/Makefile
 docs/Makefile
 docs/plugins/Makefile
 docs/version.entities
@@ -509,6 +508,9 @@
 pkgconfig/Makefile
 pkgconfig/gstreamer-plugins-libav-uninstalled.pc
 )
+if test "x$with_system_libav" != "xyes"; then
+  AC_CONFIG_FILES(gst-libs/ext/Makefile)
+fi
 AC_OUTPUT
 
 ORC_OUTPUT
